osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/main/java/org/opensmartgridplatform/adapter/protocol/dlms/application/threads/RecoverKeyProcess.java

```diff
@@ -124,65 +102,22 @@ private void checkState() {
         }
     }
 
-    private boolean canConnectUsingNewKeys() {
+    private boolean canConnectUsingNewKeys(DlmsDevice device) {
         DlmsConnection connection = null;
         try {
-            connection = this.createConnectionUsingNewKeys();
-            return true;
-        } catch (final Exception e) {
-            LOGGER.warn("Connection exception: {}", e.getMessage(), e);
+            connection = this.hls5Connector.connectUnchecked(device, null, this.secretManagementService::getNewKey);
```

Comment:
You're right, it would require some work and it isn't obvious to me how/at what point it should be added and how much 'added value' it would bring. So for now, I thought this would suffice.